pkg/controller/operators/operatocondition_controller.go

```diff
@@ -0,0 +1,259 @@
+	log.V(1).Info("reconciling operatorcondition")
+
+	operatorCondition := &operatorsv1.OperatorCondition{}
+	err := r.Client.Get(context.TODO(), req.NamespacedName, operatorCondition)
+	if err != nil {
+		log.V(1).Info("Unable to find operatorcondition", "error", err)
+		return ctrl.Result{}, err
+	}
+
+	ownerReference := ownerutil.GetOwnerByKind(operatorCondition, operatorsv1alpha1.ClusterServiceVersionKind)
+	if ownerReference == nil {
+		return ctrl.Result{}, fmt.Errorf("No CSV Owner specified")
+	}
+
+	err = r.ensureOperatorConditionRole(*operatorCondition, *ownerReference)
+	if err != nil {
+		log.V(1).Info("Error reconciling operatorcondition", "error", err)
+		return ctrl.Result{Requeue: true}, err
+	}
+
+	err = r.ensureDeploymentEnvVars(*operatorCondition, *ownerReference)
+	if err != nil {
+		log.V(1).Info("Error reconciling operatorcondition", "error", err)
+		return ctrl.Result{Requeue: true}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *OperatorConditionReconciler) ensureOperatorConditionRole(operatorCondition operatorsv1.OperatorCondition, ownerReference metav1.OwnerReference) error {
+	r.log.V(1).Info("Creating the RBAC for the operatorCondition", "operatorConditions", operatorCondition)
+
+	// get all Service Accounts
+	serviceAccountList := corev1.ServiceAccountList{}
+	err := r.Client.List(context.TODO(), &serviceAccountList, client.InNamespace(operatorCondition.GetNamespace()))
+	if err != nil {
+		return err
+	}
+
+	subjects := []rbacv1.Subject{}
+	for _, serviceAccount := range filterServiceAccounts(serviceAccountList.Items, hasOwnerFunc(ownerReference)) {
+		subjects = append(subjects, rbacv1.Subject{
+			Kind:     rbacv1.ServiceAccountKind,
+			Name:     serviceAccount.GetName(),
+			APIGroup: "",
+		})
+	}
+
+	role := rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            operatorCondition.GetName(),
+			Namespace:       operatorCondition.GetNamespace(),
+			OwnerReferences: []metav1.OwnerReference{ownerReference},
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs:         []string{"get"},
+				APIGroups:     []string{"operators.coreos.com"},
+				Resources:     []string{"operatorconditions"},
+				ResourceNames: []string{operatorCondition.GetName()},
+			},
+			{
+				Verbs:         []string{"get,update,patch"},
+				APIGroups:     []string{"operators.coreos.com"},
+				Resources:     []string{"operatorconditions/status"},
+				ResourceNames: []string{operatorCondition.GetName()},
+			},
+		},
+	}
+	err = r.Client.Create(context.TODO(), &role)
+	if err != nil {
+		if !k8serrors.IsAlreadyExists(err) {
+
+			return err
+		}
+		existingRole := rbacv1.Role{}
+		err := r.Client.Get(context.TODO(), client.ObjectKey{Name: role.GetName(), Namespace: role.GetNamespace()}, &existingRole)
+		if err != nil {
+			return err
+		}
+		existingRole.OwnerReferences = role.OwnerReferences
+		existingRole.Rules = role.Rules
+		err = r.Client.Update(context.TODO(), &existingRole)
```

Comment:
Will this generate a lot of write load if the update call is made whether or not there's a change? I'm especially concerned with the case where an operator spams writes to its condition resource and this controller amplifies that traffic.

Reply:
Good point - I'll update the code to check if any of the fields that the controller sets were modified before updating the resource.

Reply:
^applies to deps, roles, and roleBindings

Reply:
Addressed in the latest version of the PR.